### ISSUE TITLE
Gen.Set().WithCountLessThan() should trigger "best attempt" set generation

### ISSUE THREAD

**Issue by nth-commit** [nth-commit/GalaxyCheck#256]
This should be a valid generator

```csharp
Gen.Boolean().SetOf().WithCountLessThan(10)
```

However, currently it will generate a count between 0 and 10, and if that count is > 2 (max distinct booleans), it will exhaust.

It should be less strict about when constrained by max count, and just do its best.
